platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -0,0 +1,720 @@
+
+##### Consensus Bottlenecks
+
+Let us suppose that we have a node, Alice. Perhaps initially Alice is able to receive and process events at the same
+speed as other nodes in the network. Perhaps the network load increases past some point that Alice can handle.  At this
+point, other nodes are receiving, creating, gossiping, and coming to consensus on rounds faster than Alice. Remember:
+
+1. Birth-round filtering limits the number of events received by Alice to coincide with the pace at which Alice is
+   handling consensus.
+2. The Tipset algorithm only creates events when doing so will advance consensus. It relies on events that have already
+   passed through Birth-round filtering.
+
+As the other nodes progress farther than Alice, they begin to send events with newer and newer birth rounds. At some
+point, they get too far ahead and begin delivering events that are too far in the future and fail to pass Alice's
+birth-round filter. This prevents Alice from being overwhelmed by events.
+
+Further, since events are coming more slowly to Alice, she will also create her own events more slowly.
+
+Further, since events are coming more slowly to Alice, her Event Creator will update more slowly. When the tipset
+algorithm is consulted for an "other parent", it will find after a short time that there are no other parents it can
+select that will advance consensus. This will cause it to stop creating events until enough events from other creators
+have been processed. This natural slowing of event creation provides a virtuous cycle: as each stressed node slows down
+the event creation rate, it starts to create events with more and more transactions within each event and at a slower
+rate. This will lead to fewer overall events, allowing it to do less work. If the node is still not able to keep up,
+eventually it will refuse to accept any additional transactions from users. If enough nodes in the network are stressed,
+then the overall transaction ingestion rate of the network will be reduced, further reducing the amount of work each
+node has to do. Eventually, an equilibrium is reached.
+
+If the rate at which the network is creating events slows, Alice will be able to catch up by retrieving those previously
+dropped events through gossip, and will be able to process them and catch up. Or, in the last extremity where Alice has
+fallen too far behind, Alice will wait for some time and reconnect.
+
+In addition, the Gossip module reports to the Execution layer health information about each node in the network, as
+well as its own health. Health in this case can simply be the highest birth-round received per node in the network. The
+birth-round is a good proxy for how healthy the node is, since slower, unhealthy nodes will have lower birth-rounds
+relative to their peers. Using this information, the Execution layer can throttle transaction ingestion network-wide.
+How this is done will be described in the "dynamic throttles" design documentation.
```

Comment:
OK, it took me a minute to understand what the argument was. At no time did we think we'd be using information from < 1/3 of the nodes to make a decision (we'd have taken health information from all nodes and kept only 2/3 + us, and then found the median and calculated health from that point).

But actually, this doesn't make sense as you point out, because the hashgraph can only progress at > 2/3 consensus, and that means > 2/3 of us are on some "recent" birth round. So the only people we'll be getting old birth rounds from are the < 1/3, so either that information is useless, or we are paying attention to the bad nodes.

And since we'd never pay attention to the bad nodes, it meant the information is useless.

So we should eliminate this paragraph. Other than, we should say it is up to the execution layer to determine health of nodes and adjust throttles accordingly, as per the dynamic throttles design which is TBD.

Reply:
I will remove the paragraph 👍

Reply:
Sounds good. Absent this, remaining points I brought up would probably amount to a -0.5 vote.

Reply:
I've either fixed or responded to all of your comments